rust/actyx/ax-core/src/runtime/eval.rs

```diff
@@ -145,6 +153,18 @@ impl<'a> Context<'a> {
 
     pub fn eval_from<'b, 'c: 'b>(&'b self, expr: &'c TagExpr) -> BoxFuture<'b, anyhow::Result<Cow<'c, TagExpr>>> {
         async move {
+            // Below block sets a threshold for the remaining stack during
+            // heavily recursive AQL evaluation to avoid crashing the whole
+            // databank process.
+            if let Some(x) = stacker::remaining_stack() {
+                if x < EVAL_REMAINING_STACK_THRESHOLD {
+                    return Err(anyhow!(
```

Comment:
I’d use `anyhow::bail!` in such cases.

Reply:
is there any difference between them other than one is a macro-shortcut for the other?

Reply:
no difference, it is just shorter, so easier to read

Reply:
ah ok. done